code/datums/components/greentext.dm

```diff
@@ -0,0 +1,50 @@
+/datum/component/greentext
+	var/mob/living/last_holder = null
+	var/mob/living/current_holder
+	var/list/color_altered_mobs = list()
+	var/datum/callback/roundend_callback
+	var/quiet = FALSE
+
+/datum/component/greentext/Initialize()
+	if (!isitem(parent))
+		return COMPONENT_INCOMPATIBLE
+	RegisterSignal(parent, COMSIG_ITEM_PICKUP, .proc/equip)
+	RegisterSignal(parent, COMSIG_ITEM_DROPPED, .proc/unequip)
+	roundend_callback = CALLBACK(src, .proc/check_winner)
+	SSticker.OnRoundend(roundend_callback)
+
+/datum/component/greentext/proc/equip(item , mob/living/user as mob)
```

Comment:
what? why? who? IS TOM BACK FOR BLOOD?

Reply:
you never use the item arg, this can just be
```suggestion
/datum/component/greentext/proc/equip(mob/living/user)
```

Reply:
@Fikou  I don't think that works. The first arg is always the thing the signal is coming from.

Reply:
he doesn't need the as bit tho